sketch-on-map/src/main/java/com/esri/arcgismaps/sample/sketchonmap/MainActivity.kt

```diff
@@ -0,0 +1,299 @@
+     * selected, and start multipoint drawing mode.
+     */
+    fun createModeMultipoint(view: View) {
+        setCurrentSelectionText("Multipoint")
+        resetButtons()
+        activityMainBinding.pointLinePolygonToolbar.multipointButton.isSelected = true
+        geometryEditor?.apply {
+            tool = vertexTool
+            start(GeometryType.Multipoint)
+        }
+    }
+
+    /**
+     * When the polyline button is clicked, reset other buttons, show the polyline button as
+     * selected, and start polyline drawing mode.
+     */
+    fun createModePolyline(view: View) {
+        setCurrentSelectionText("Polyline")
+        resetButtons()
+        activityMainBinding.pointLinePolygonToolbar.polylineButton.isSelected = true
+        geometryEditor?.apply {
+            tool = vertexTool
+            start(GeometryType.Polyline)
+        }
+    }
+
+    /**
+     * When the polygon button is clicked, reset other buttons, show the polygon button as
+     * selected, and start polygon drawing mode.
+     */
+    fun createModePolygon(view: View) {
+        setCurrentSelectionText("Polygon")
+        resetButtons()
+        activityMainBinding.pointLinePolygonToolbar.polygonButton.isSelected = true
+        geometryEditor?.apply {
+            tool = vertexTool
+            start(GeometryType.Polygon)
+        }
+    }
+
+    /**
+     * When the freehand line button is clicked, reset other buttons, show the freehand line
+     * button as selected, and start freehand line drawing mode.
+     */
+    fun createModeFreehandLine(view: View) {
+        setCurrentSelectionText("FreehandPolyline")
+        resetButtons()
+        activityMainBinding.pointLinePolygonToolbar.freehandLineButton.isSelected = true
+        geometryEditor?.apply {
+            tool = freehandTool
+            start(GeometryType.Polyline)
+        }
+    }
+
+    /**
+     * When the freehand polygon button is clicked, reset other buttons, show the freehand
+     * polygon button as selected, and enable freehand polygon drawing mode.
+     */
+    fun createModeFreehandPolygon(view: View) {
+        setCurrentSelectionText("FreehandPolygon")
+        resetButtons()
+        activityMainBinding.pointLinePolygonToolbar.freehandPolygonButton.isSelected = true
+        geometryEditor?.apply {
+            tool = freehandTool
+            start(GeometryType.Polygon)
+        }
+    }
+
+    /**
+     * When the undo button is clicked, undo the last event on the GeometryEditor.
+     */
+    fun undo(view: View) {
+        if (geometryEditor?.canUndo?.value == true) {
+            geometryEditor?.undo()
+            setCurrentSelectionText(getString(R.string.undo))
+        }
+    }
+
+    /**
+     * When the redo button is clicked, redo the last undone event on the GeometryEditor.
+     */
+    fun redo(view: View) {
+        if (geometryEditor?.canRedo?.value == true) {
```

Comment:
The `canUndo` and `canRedo` properties are listenable so if you wanted to you could listen to them and disable/enable those buttons there instead of doing these checks